### DISCUSSION TITLE
How to load a model in bf16 format

### DISCUSSION THREAD

**Discussion by georgel2004** [InternLM/lmdeploy#1540] [General]
In turbomind.py the data_type is hard coded to fp16. 
If I change the value to bf16 I could see increase in performance speed. 
How could I load a model in bf16 format using the pipeline?

**Comment by lvhan028** [comment]
turbomind doesn't hard code `data_type` fp16. It supports bf16 if `torch_dtype` in model's config.json is bf16 and the target device supports bf16 as well
https://github.com/InternLM/lmdeploy/blob/7c4e75b53a8c891748fb7639c06e1ac20913775d/lmdeploy/turbomind/turbomind.py#L271
